src/vse_sync_pp/analyzers/pmc.py

```diff
@@ -0,0 +1,147 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Analyze PMC log messages"""
+
+from .analyzer import Analyzer
+
+STATE_FREERUN = 248
+STATE_LOCKED = 6
+STATE_HOLDOVER_IN_SPEC = 7
+STATE_HOLDOVER_OUT_OF_SPEC1 = 140
+STATE_HOLDOVER_OUT_OF_SPEC2 = 150
+STATE_HOLDOVER_OUT_OF_SPEC3 = 160
+
+def handle_freerun_transition(clock_state_analyzer, clock_class):
+    """Transition from the freerun state"""
+    try:
+        clock_state_analyzer.state = FREERUN_TRANSITION[clock_class]
+    except KeyError:
+        return False
+    
+def handle_locked_transition(clock_state_analyzer, clock_class):
+    """Transition from the locked state"""
+    try:
+        clock_state_analyzer.state = LOCKED_TRANSITION[clock_class]
+    except KeyError:
+        return False
+    
+def handle_holdover_in_spec_transition(clock_state_analyzer, clock_class):
+    """Transition from the holdover in spec state"""
+    try:
+        clock_state_analyzer.state = HOLDOVER_IN_SPEC_TRANSITION[clock_class]
+    except KeyError:
+        return False
+    
+def handle_holdover_out_of_spec_transition(clock_state_analyzer, clock_class):
+    """Transition from the holdover out of spec state"""
+    try:
+        clock_state_analyzer.state = HOLDOVER_OUT_OF_SPEC_TRANSITION[clock_class]
+    except KeyError:
+        return False
+
+STATE_TRANSITION = {
+    STATE_FREERUN: handle_freerun_transition,
+    STATE_LOCKED: handle_locked_transition,
+    STATE_HOLDOVER_IN_SPEC: handle_holdover_in_spec_transition,
+    STATE_HOLDOVER_OUT_OF_SPEC1: handle_holdover_out_of_spec_transition,
+    STATE_HOLDOVER_OUT_OF_SPEC2: handle_holdover_out_of_spec_transition,
+    STATE_HOLDOVER_OUT_OF_SPEC3: handle_holdover_out_of_spec_transition,
+}
+
+FREERUN_TRANSITION = {
+    STATE_FREERUN: STATE_FREERUN,
+    STATE_LOCKED: STATE_LOCKED,
+}
+
+LOCKED_TRANSITION = {
+    STATE_LOCKED: STATE_LOCKED,
+    STATE_HOLDOVER_IN_SPEC: STATE_HOLDOVER_IN_SPEC,
+}
+
+HOLDOVER_IN_SPEC_TRANSITION = {
+    STATE_LOCKED: STATE_LOCKED,
+    STATE_HOLDOVER_IN_SPEC: STATE_HOLDOVER_IN_SPEC,
+    STATE_HOLDOVER_OUT_OF_SPEC1: STATE_HOLDOVER_OUT_OF_SPEC1,
+    STATE_HOLDOVER_OUT_OF_SPEC2: STATE_HOLDOVER_OUT_OF_SPEC2,
+    STATE_HOLDOVER_OUT_OF_SPEC3: STATE_HOLDOVER_OUT_OF_SPEC3,
+}
+
+HOLDOVER_OUT_OF_SPEC_TRANSITION = {
+    STATE_LOCKED: STATE_LOCKED,
+    STATE_HOLDOVER_OUT_OF_SPEC1: STATE_HOLDOVER_OUT_OF_SPEC1,
+    STATE_HOLDOVER_OUT_OF_SPEC2: STATE_HOLDOVER_OUT_OF_SPEC2,
+    STATE_HOLDOVER_OUT_OF_SPEC3: STATE_HOLDOVER_OUT_OF_SPEC3,
+}
```

Comment:
```suggestion
HOLDOVER_OUT_OF_SPEC_CAT1_TRANSITION = {
    STATE_LOCKED: STATE_LOCKED,
    STATE_HOLDOVER_OUT_OF_SPEC_CAT1: STATE_HOLDOVER_OUT_OF_SPEC_CAT1,
}

HOLDOVER_OUT_OF_SPEC_CAT2_TRANSITION = {
    STATE_LOCKED: STATE_LOCKED,
    STATE_HOLDOVER_OUT_OF_SPEC_CAT2: STATE_HOLDOVER_OUT_OF_SPEC_CAT2,
}

HOLDOVER_OUT_OF_SPEC_CAT3_TRANSITION = {
    STATE_LOCKED: STATE_LOCKED,
    STATE_HOLDOVER_OUT_OF_SPEC_CAT3: STATE_HOLDOVER_OUT_OF_SPEC_CAT3,
}```

Reply:
Is there any difference in the out-of-spec transitions? why it couldn't be the same?

Reply:
We cannot have a transition like: OUT_OF_SPEC_CAT1 -->  OUT_OF_SPEC_CAT3